### PR TITLE
Add short descriptions for each DB worker

### PR DESCRIPTION
**Pull request by vdavid**
- This PR accompanies my larger effort to create a dashboard for our background jobs: https://github.com/sourcegraph/sourcegraph/pull/44901

I used the info from the [workers docs page](https://docs.sourcegraph.com/admin/workers) and my [spreadsheet](https://docs.google.com/spreadsheets/d/1Ck9LVsFqEB1ukgE2kziYTskTYis1c44wuxcyzIqg5Tw/edit#gid=0) to add a short description for each DB-backed worker.

Before merging this PR, I'd like to give an opportunity to everyone who owns workers to make the descriptions better. In some cases, I found very little about what a worker did, so I made my best guess.

We'll display the descriptions similarly to this (descriptions highlighted):

CleanShot 2023-01-13 at 14 15 13@2x

I tried not to repeat what was obvious from the worker's name and to be concise and use a similar style for each item.
However, the pre-existing descriptions for periodic workers that are also on the list are pretty diverse in style, so the bar is not very high.

## Test plan

Check if everyone's happy with the descriptions

### CODE REVIEW

File: enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_indexing_scheduler.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ func NewDependencyIndexingScheduler(
 
 	return dbworker.NewWorker[shared.DependencyIndexingJob](rootContext, dependencyIndexingStore, handler, workerutil.WorkerOptions{
 		Name:              "precise_code_intel_dependency_indexing_scheduler_worker",
+		Description:       "queues indexing jobs for a remote executor instance to perform",
```
Comment (Strum355) [inline enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_indexing_scheduler.go:58]:
```suggestion
		Description:       "queues code-intel auto-indexing jobs for dependency packages",
```

File: enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_sync_scheduler.go [hunk 2]
```diff
@@ -45,6 +45,7 @@ func NewDependencySyncScheduler(
 
 	return dbworker.NewWorker[shared.DependencySyncingJob](rootContext, dependencySyncStore, handler, workerutil.WorkerOptions{
 		Name:              "precise_code_intel_dependency_sync_scheduler_worker",
+		Description:       "periodically checks for dependency packages that can be auto-indexed",
```
Comment (Strum355) [inline enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_sync_scheduler.go:48]:
```suggestion
		Description:       "reads dependency package references from code-intel uploads to be synced to the instance",
```

File: enterprise/internal/codeintel/uploads/internal/background/job_worker_handler.go [hunk 3]
```diff
@@ -63,6 +63,7 @@ func NewUploadProcessorWorker(
 
 	return dbworker.NewWorker(rootContext, workerStore, handler, workerutil.WorkerOptions{
 		Name:                 "precise_code_intel_upload_worker",
+		Description:          "processes precise code intel uploads",
```
Comment (Strum355) [inline enterprise/internal/codeintel/uploads/internal/background/job_worker_handler.go:66]:
```suggestion
		Description:          "processes precise code-intel uploads",
```

I think this is the way we usually reference code-intel, but im not 100% sure lol

**Comment by leonore** [comment]
you're missing a description for the new [data retention worker for code insights](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/internal/insights/background/retention/worker.go?L77). a suitable description could be "archives code insights data points over the maximum sample size". I forgot to add the worker to the docs, I'll do that now

**Comment by vdavid** [comment]
Thanks, @leonore, I've added it!

**Comment by jhchabran** [comment]
@vdavid thanks for taking the time to add those descriptions, it's so damn useful when you're not working with these often and still cross them when investigating an incident 🙏

File: enterprise/internal/insights/background/queryrunner/worker.go [hunk 4]
```diff
@@ -46,6 +46,7 @@ func NewWorker(ctx context.Context, logger log.Logger, workerStore *workerStoreE
 
 	options := workerutil.WorkerOptions{
 		Name:              "insights_query_runner_worker",
+		Description:       "runs code insights queries",
```
Comment (chwarwick) [inline enterprise/internal/insights/background/queryrunner/worker.go:49]:
```suggestion
		Description:       "runs code insights queries for daily snapshots and new recordings",
```

File: enterprise/internal/insights/scheduler/backfill_state_new_handler.go [hunk 5]
```diff
@@ -67,6 +67,7 @@ func makeNewBackfillWorker(ctx context.Context, config JobMonitorConfig) (*worke
 
 	worker := dbworker.NewWorker(ctx, workerStore, workerutil.Handler[*BaseJob](&task), workerutil.WorkerOptions{
 		Name:              name,
+		Description:       "handles backfills that are in the \"new\" state",
```
Comment (chwarwick) [inline enterprise/internal/insights/scheduler/backfill_state_new_handler.go:70]:
```suggestion
		Description:       "determines the repos for a code insight and an approximate cost of the backfill",
```

**Review by sashaostrikov — APPROVED** [review]

File: enterprise/cmd/worker/internal/permissions/bitbucket_projects.go [hunk 6]
```diff
@@ -357,6 +357,7 @@ func newBitbucketProjectPermissionsWorker(ctx context.Context, observationCtx *o
 
 	options := workerutil.WorkerOptions{
 		Name:              "explicit_permissions_bitbucket_projects_jobs_worker",
+		Description:       "reads the `explicit_permissions_bitbucket_projects_jobs` table and executes the jobs",
```
Comment (eseliger) [inline enterprise/cmd/worker/internal/permissions/bitbucket_projects.go:360]: that feels overly technical - should we make it a bit more feature-centric?

**Review by eseliger — COMMENTED** [review]
sorry for being late to the party!

File: enterprise/cmd/worker/internal/permissions/bitbucket_projects.go (hunk 6 above)
Comment (vdavid) [reply enterprise/cmd/worker/internal/permissions/bitbucket_projects.go:360]: @eseliger It does! Give me your best desc and I'll happily add it! :)

Comment (eseliger) [reply enterprise/cmd/worker/internal/permissions/bitbucket_projects.go:360]: > Syncs Bitbucket Projects via Explicit Permissions API

From @sashaostrikov :)

**Pull request merged**